workshops/test/test_host.py

```diff
@@ -47,3 +47,23 @@ def test_host_delete(self):
 
             with self.assertRaises(Host.DoesNotExist):
                 Host.objects.get(domain=host_domain)
+
+    def test_host_invalid_chars_in_domain(self):
+        """Ensure users can't put wrong characters in the host's domain field.
+
+        Invalid characters are any that match `[^\w\.-]+`, ie. domain is
+        allowed only to have alphabet-like chars, dot and dash.
+
+        The reason for only these chars lies in `workshops/urls.py`.  The regex
+        for the host_details URL has `[\w\.-]+` matching...
```

Comment:
Not sure if that's a good idea, since we have two different regexes (see https://github.com/swcarpentry/amy/pull/482/files#r38395883).